src/services/BiometricAuth.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * @author Arthur Martello <arthur.martello@etu.utc.fr>
+ *
+ * @copyright Copyright (c) 2019, SiMDE-UTC
+ * @license GPL-3.0
+ */
+
+import React from 'react';
+import { Modal, Platform, Text, View } from 'react-native';
+import * as LocalAuthentication from 'expo-local-authentication';
+import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';
+import * as Haptics from 'expo-haptics';
+import { BlurView } from 'expo-blur';
+import { _, BiometricAuth as t } from '../utils/i18n';
+import colors from '../styles/colors';
+import BlockTemplate from '../components/BlockTemplate';
+
+export const defaultSecurity = ['transfer', 'refill', 'badge-locking'];
+export const advancedSecurity = ['transfer', 'refill', 'badge-locking', 'app-opening'];
+
+export default class BiometricAuth extends React.PureComponent {
+	static async hasHardware() {
+		return LocalAuthentication.hasHardwareAsync();
+	}
+
+	constructor(props) {
+		super(props);
+		this.state = {
+			showModal: false,
+			accepted: false,
+		};
+	}
+
+	async authenticate(successCallback, errorCallback, message = t('default_message')) {
+		const { restrictions, action } = this.props;
+
+		if (action != null && !restrictions.includes(action)) {
+			successCallback();
+			return;
+		}
+
+		const hasHardware = await BiometricAuth.hasHardware();
+		if (!hasHardware) {
+			successCallback();
```

Comment:
How having no biometric hardware (when we enable the security policy) returns a successful response ?

Reply:
A bit hard to imagine it from the first reading 😅
If you don't have any hardware, you are always authorized to do the action.
Else, the authenticate function will block you and ask for your finger.

Reply:
Then, I think, you have to wipe all data, because the security policy was compromised

Reply:
No you can't do that! `authorize()` is called for every actions, if you don't have any  hardware, you must always be authorized.
Otherwise, the security is checked every time you open the app. (And wipe if the hardware has been disabled)

Reply:
With a root, you can disable an hardware interface at a moment or another. So, I can bypass your checks, if the app  was already launched

Reply:
Should be fix with the last commit :)